Fix #5606: Logic Proof Truncation and Display

ScottJRoberts opened the pull request:
# Explanation
Fixes #5606:

This PR sets the short answer output of the logic proof interaction to be truncated if it contains multiple lines. This was done by editing the output of the `file logic_proof_short_response_directive.html` to truncate the entire joined output from the answer instead of only using the erroneous line.

Also, set the popup displayed when a short response is clicked to show the entire response and not the empty popup.

Commit:
fix 5606 logic proof response display

seanlip commented:
Thanks @ScottJRoberts! Assigning @aks681 for review.

DubeySandeep commented:
Hi @ScottJRoberts, Sorry to say but we already have one PR for #5606 at #5797 and we can only have one PR for an issue. Would you mind picking up any different isssue (see [here](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#finding-something-to-do))?
In general please make a comment in the issue thread before starting working on it, that way it becomes easier for anyone two understands that you are working on that issue and we will never fall in this kinda case.  
/cc @aks681

MShaker commented:
@DubeySandeep I deleted the other PR because my local machine was having trouble running Oppia properly on Windows. I submitted this PR from @ScottJRoberts machine to get it completed faster.

DubeySandeep commented on extensions/interactions/LogicProof/directives/logic_proof_response_directive.html:
```diff
@@ -1,5 +1,5 @@
 <div>
-  <[answer.displayed_question]>
+  <p>Assuming <[answer.assumptions_string]>; prove <[answer.target_string]>.</p>
```
Can we use `I18N_INTERACTIONS_LOGIC_PROOF_QUESTION_STR_ASSUMPTIONS` here?

ScottJRoberts replied:
I switched to using `I18N_INTERACTIONS_LOGIC_PROOF_QUESTION_STR_ASSUMPTIONS`

DubeySandeep commented on extensions/interactions/LogicProof/directives/logic_proof_short_response_directive.html:
```diff
@@ -4,5 +4,5 @@
 </span>
 <span ng-if="!answer.correct">
   <span translate="I18N_INTERACTIONS_LOGIC_PROOF_INCORRECT_ANSWERS"></span>:
-  <pre><[answer.displayed_proof[1] | truncateAtFirstLine]></pre>
```
Do we have to do something like this for `ng-if="answer.correct"`?

MShaker replied:
I don't think so actually. It is an interesting setup, but the way it works are that correct answers are sent as strings, while incorrect answers are arrays of 3 strings where the middle value is the erroneous line. That's why truncation wasn't working, because only the line with the error was being displayed on wrong answers.

DubeySandeep replied:
Thanks for checking!

DubeySandeep reviewed (changes requested):
Hi @ScottJRoberts, Thanks for the PR, from UI perspective it looks as expected. I have left few inline comments PTAL!

Also, the popup looks weird for long answers, can we make it verically scroll? 
![image](https://user-images.githubusercontent.com/16653571/48462356-da4cab80-e7fd-11e8-936b-e907834c8cba.png)

Commit:
added scroll and corrected assumption sting for logic proof response

ScottJRoberts commented:
@DubeySandeep in the latest commit I switched to using `I18N_INTERACTIONS_LOGIC_PROOF_QUESTION_STR_ASSUMPTIONS`. I also just added styling to the div containing the response so that it has a max height and scrolls if there is overflow. I am not sure if there is better method that you had in mind for implementing the scroll. Please let me know what you think

DubeySandeep commented on core/templates/dev/head/pages/exploration_player/input_response_pair_directive.html:
```diff
@@ -71,8 +71,8 @@
 <script type="text/ng-template" id="popover/answer">
-  <div>
-    <angular-html-bind html-data="answer()">
+  <div style="max-height:300px;overflow-y:scroll;">
```
Make it `style="max-height: 300px; overflow-y: scroll;"`

DubeySandeep commented on extensions/interactions/LogicProof/directives/LogicProof.js:
```diff
@@ -225,7 +225,7 @@ oppia.directive('oppiaInteractiveLogicProof', [
               proof_string: $scope.proofString,
               proof_num_lines: $scope.proofString.split('\n').length,
-              displayed_question: $scope.questionString
+              displayed_question: $scope.questionString,
```
remove `,`.

ScottJRoberts replied:
I removed this

DubeySandeep commented on core/templates/dev/head/pages/exploration_player/input_response_pair_directive.html:
```diff
@@ -71,8 +71,8 @@
 <script type="text/ng-template" id="popover/answer">
-  <div>
-    <angular-html-bind html-data="answer()">
+  <div style="max-height:300px;overflow-y:scroll;">
```
Make it `style="max-height: 300px; overflow-y: scroll;"`

ScottJRoberts replied:
I spaced the styling here

DubeySandeep reviewed (commented):
@ScottJRoberts, Changes LGTM, just a few nits required then it will be good to go.

Commit:
completed requested style changes

DubeySandeep reviewed (approved):
@ScottJRoberts, Thanks for the changes, LGTM!
